cidc_api/resources/downloadable_files.py

```diff
@@ -80,27 +80,26 @@ def get_filelist(args):
     Return a file `filelist.tsv` mapping GCS URIs to flat filenames for the
     provided set of file ids.
     """
-    # Build query filter
+    # Build the permissions filter
     current_user = get_current_user()
     user_perms_filter = DownloadableFiles.build_file_filter(user=current_user)
-    file_ids = args["file_ids"]
-    batch_id_filter = lambda q: user_perms_filter(
-        q.filter(DownloadableFiles.id.in_(file_ids))
-    )
 
-    # Get requested files
-    files = DownloadableFiles.list(filter_=batch_id_filter)
+    # Get request object_urls
+    urls = DownloadableFiles.list_object_urls(
+        args["file_ids"], filter_=user_perms_filter
+    )
```

Comment:
This fixes a relevant bug with the `/downloadable_files/filelist` endpoint, where filelist files were getting truncated by the default pagination functionality of `DownloadableFiles.list`.